cmd/download.go

```diff
@@ -264,6 +197,119 @@ func (d download) needsSlugWhenGivenTrackOrTeam() error {
 	return nil
 }
 
+func (d download) metadata() *ws.ExerciseMetadata {
+	return &ws.ExerciseMetadata{
+		AutoApprove:  d.Solution.Exercise.AutoApprove,
+		Track:        d.Solution.Exercise.Track.ID,
+		Team:         d.Solution.Team.Slug,
+		ExerciseSlug: d.Solution.Exercise.ID,
+		ID:           d.Solution.ID,
+		URL:          d.Solution.URL,
+		Handle:       d.Solution.User.Handle,
+		IsRequester:  d.Solution.User.IsRequester,
+	}
+}
+
+func (d download) exercise() ws.Exercise {
+	return d.metadata().Exercise(d.workspace)
+}
+
+// requestFile requests a Solution file from the API, returning an HTTP response.
+// 0 Content-Length responses are swallowed, returning nil.
+func (d download) requestFile(filename string) (*http.Response, error) {
```

Comment:
I could see this being its own function not tied to download. Maybe this is something that should live on the Exercism Client API with a single argument being the `parsedURL.string()` or if usign @kytrinyx Solution File rerfactor `sf.url()`.

Reply:
I like the idea of having a helper on the client API with a single URL argument. That feels like a perfectly reasonable (and cohesive!) bit of behavior for the CLI.